src/util/simplify_expr_int.cpp

```diff
@@ -446,6 +446,8 @@ bool simplify_exprt::simplify_mod(exprt &expr)
 
 bool simplify_exprt::simplify_plus(exprt &expr)
 {
+  // only used to check structural invariants
+  (void)to_plus_expr(expr);
```

Comment:
Maybe still use this, in particular the const version thereof, in the next lines? Doing `.type()` on a `const` expression is cheaper, thus beneficial.